libfreerdp/core/server.c

```diff
@@ -96,10 +96,10 @@ static void wts_queue_receive_data(rdpPeerChannel* channel, const BYTE* Buffer,
 	buffer = (BYTE*) (messageCtx + 1);
 	CopyMemory(buffer, Buffer, Length);
 
-	MessageQueue_Post(channel->queue, messageCtx, 0, NULL, NULL);
+	return MessageQueue_Post(channel->queue, messageCtx, 0, NULL, NULL);
```

Comment:
not your code but since this function is bool now you could also check the malloc 7 lines above and return false if it failed.